2008-08-16  Adrien DELLE CAVE <adellecave@proformatique.com>

	pf_iax2_moh-config.patch
	$Revision$
	$Date$
	Bugfix

	Wrong variables used like size reference for mohinterpret and mohsuggest
	in channel IAX2's configuration.

Index: asterisk-1.4.21/channels/chan_iax2.c
===================================================================
--- asterisk-1.4.21.1.orig/channels/chan_iax2.c	2008-06-03 23:34:55.000000000 +0200
+++ asterisk-1.4.21.1/channels/chan_iax2.c	2008-08-16 11:07:58.000000000 +0200
@@ -10043,9 +10043,9 @@ static int set_config(char *config_file,
 		} else if (!strcasecmp(v->name, "accountcode")) {
 			ast_copy_string(accountcode, v->value, sizeof(accountcode));
 		} else if (!strcasecmp(v->name, "mohinterpret")) {
-			ast_copy_string(mohinterpret, v->value, sizeof(user->mohinterpret));
+			ast_copy_string(mohinterpret, v->value, sizeof(mohinterpret));
 		} else if (!strcasecmp(v->name, "mohsuggest")) {
-			ast_copy_string(mohsuggest, v->value, sizeof(user->mohsuggest));
+			ast_copy_string(mohsuggest, v->value, sizeof(mohsuggest));
 		} else if (!strcasecmp(v->name, "amaflags")) {
 			format = ast_cdr_amaflags2int(v->value);
 			if (format < 0) {
